p2p/src/sync/peer.rs

```diff
@@ -317,6 +320,26 @@ where
     async fn handle_header_list(&mut self, headers: Vec<SignedBlockHeader>) -> Result<()> {
         log::debug!("Headers list from peer {}", self.id());
 
+        if let Some(last_header) = headers.last() {
```

Comment:
Please move this block to a separate function, let's call it `wait_for_clock_diff`.